File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int t = Arrays.stream(nums).sum() - x;

        if (t == 0) {
            return nums.length;
        }

        int n = nums.length;
        int l = 0;
        int c = 0;
        int m = Integer.MAX_VALUE;

        for (int r = 0; r < n; r++) {
            c += nums[r];

            while (c > t && l <= r) {
                c -= nums[l];
                l += 1;
            }

            if (c == t) {
                m = Math.min(m, n - (r - l + 1));
            }
        }

        if (m == Integer.MAX_VALUE) {
            return -1;
        } else {
            return m;
        }
    }
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:        
        t = sum(nums) - x

        if t == 0:
            return len(nums)

        n = len(nums)
        l = 0
        c = 0
        m = float('inf')

        for r in range(n):
            c += nums[r]

            while c > t and l <= r:
                c -= nums[l]
                l += 1

            if c == t:
                m = min(m, n - (r - l + 1))

        if m == float('inf'):
            return -1
        else:
            return m 
